tokio-trace/tests/filter_caching_is_lexically_scoped.rs

```diff
@@ -0,0 +1,64 @@
+// Tests that depend on a count of the number of times their filter is evaluated
+// cant exist in the same file with other tests that add subscribers to the
+// registry. The registry was changed so that each time a new dispatcher is
+// added all filters are re-evaluated. The tests being run only in separate
+// threads with shared global state lets them interfere with eachother
```

Comment:
it's pretty sad that we have to do this, but it feels like a better solution than `RUST_TEST_THREADS=1`... 👍